### DISCUSSION TITLE
Wildcards in Lists

### DISCUSSION THREAD

**Discussion by always-depends** [kkapsner/CanvasBlocker#546] [Q&A] [answered]
I tried using wildcards in the Canvas Blocker whitelist and it doesn't seem to be working.  Can you confirm if wildcards are supported or not?

Some examples of items I would like to whitelist:
- Either/Or: My devices run with the `.lan` TLD, or `10.*.*.*`, other private IP ranges
- `127.0.0.*`

Thanks!

**Comment by kkapsner** [comment, marked as answer]
Wildcards are not supported in CanvasBlocker. But it supports regular expressions. So you could use `^https?://10\.\d+\.\d+\.\d` or `^https?://[^/]+\.lan`.

**Reply by always-depends** [reply]
That will work.  Thanks a lot!

**Comment by GCRaistlin** [comment]
This makes CF active only for foo.com (and its subdomains) and sub.bar.com:
`Whitelist: ^https?://(?!([^/]+?\.)?foo\.com)(?!sub\.bar\.com)`
I believe is is worth to add this to the Whitelist's tip.

**Reply by kkapsner** [reply]
If you want to enable CB only for certain domains you should do it the other way around: set the block mode to "allow everything" and add an URL specific value for the domains where it should be active and set the mode to "fake".
